tensorboard/webapp/widgets/data_table/column_selector_component.ts

```diff
@@ -93,12 +93,12 @@ export class ColumnSelectorComponent implements OnInit, AfterViewInit {
 
   ngAfterViewInit() {
     this.searchInput = '';
-    this.searchField.nativeElement.focus();
     this.selectedIndex$.next(0);
-  }
-
-  focus() {
-    this.searchField?.nativeElement.focus();
+    this.activate();
+    // Wait until next tick to prevent https://angular.io/errors/NG0100
```

Comment:
I still got these errors, though, when adding new columns using the selector.

![image](https://github.com/tensorflow/tensorboard/assets/17152369/c23e0488-6043-42a0-9a5a-67f595118d9f)

Reply:
Resolved by passing the calling component's ViewContainerRef to `createNextToElement()`.

It was caused by the template values in the **app root** ViewContainer being updated after the DataTableComponent template was checked. Using the calling Component's own ViewContainer ensures proper change detection order without any intervention.